### DISCUSSION TITLE
Get the repository information without winget

### DISCUSSION THREAD

**Discussion by Mannshoch** [microsoft/winget-pkgs#188872] [Q&A] [answered]
We have the idea to create update notifications of particular software from our Monitoring tools. Is there a way to get the catalog e.g. from a RestAPI.

**Comment by SpecterShell** [comment]
https://cdn.winget.microsoft.com/cache/source2.msix

**Reply by Mannshoch** [reply]
Do you have an example how to use this source correctly?

**Comment by vedantmgoyal9** [comment]
@Mannshoch you can use https://vedantmgoyal.vercel.app/api/winget-pkgs/versions/*. You can replace the `*` with any package identifier if you don't want all packages. If you want information other than packages and their versions, you can tell me and I will look into if it can be made accessible through the aforementioned endpoint, or I can also setup another one.

**Reply by Mannshoch** [reply]
Thank you very much for the offer @vedantmgoyal9 
Even Winget as a source of information about application updates is not a popular idea among us. (If possible, we would rather have something from the application manufacturer.) Unfortunately it is unlikely that we will use another third party source to get update information from Winget.
It is more likely that we will run a Winget proxy ourselves. If it can be realized via a script language like powershell or python. (we have no programming skills but a little script experience)
We are also considering whether we could parse the output of Winget list and write it to a JSON. (maybe I'll write a feature request for this)

**Reply by vedantmgoyal9** [reply, marked as answer]
@Mannshoch `winget list` is for listing the installed programs on a machine. I don't think you can use it as a source for determining app updates.

Also, there is `winget show --versions <package-id>` which outputs all versions of a package. You can then execute `winget show <package-id> -v <version>` to get more information like download url, publisher, description, homepage, etc... You can also use the WinGet PowerShell module to get output as `PSObject` and the pipe them to `ConvertTo-Json` to get output in JSON format.

**Reply by Mannshoch** [reply]
Thank you. I will try the powershell.
Do I understand it correct, that this repository develop both a winget version for cmd and for powershell but the Microsoft store app "Appinstaller" only ship the cmd version?

**Reply by vedantmgoyal9** [reply]
@Mannshoch yes, you would have to install the PowerShell module manually. Also, the PowerShell module requires winget installed. Without it, PowerShell module will just fail to run.
